web/packages/teleport/src/teleportContext.tsx

```diff
@@ -104,7 +104,7 @@ class TeleportContext implements types.Context {
       !storageService.getOnboardDiscover()
     ) {
       const hasResource =
-        await userService.checkUserHasAccessToRegisteredResource();
+        await userService.checkUserHasAccessToAnyRegisteredResource();
```

Comment:
Just another reminder that since we are using the unified resource API,  we can remove this redundant call and move this block out from teleportContext and put it inside unified resource view. 

It hits the API twice otherwise, fetching same resource. 

In the unified view, this will be just:
```
if (
      this.storeUser.hasPrereqAccessToAddAgents() &&
      this.storeUser.hasAccessToQueryAgent() &&
      !storageService.getOnboardDiscover()
    ) {
      const hasResource = unifiedresourece_count === 0;
      storageService.setOnboardDiscover({ hasResource });
 }
```

Reply:
but what if the user is taken to a route that is not the unified resources view upon login?

Reply:
Showing onboarding modal on other route will be a bad user experience no? 

Say the admin is trying to add a new user or role and we pop up modal asking to add a resource, I do not think that will be a good greeting.

Reply:
> Showing onboarding modal on other route will be a bad user experience no?

hm, maybe it is a bit invasive. i saw it as like "hey, remember, we have this resource feature 😃"
we can ask in design to confirm